_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,658 @@
+</article>
+<article><center>
+
+
+| Project   |  What it does  | Why it's awesome |
+|----------|-------------|------|
+| [Babel][] | Transform source code | You can pick and choose language features |
+| [TypeScript][] | Transform source code | Microsoft choose all your language features |
+| [NPM][] | CocoaPods for JS | Make your own standard library |
+| [Yarn][] | Bundler for JS | An NPM compatible re-think of the NPM CLI |
+| [ESLint][] | Static code linter | Makes it hard to write bad code in JS |
+| [TSLint][] | Static code linter | Makes it hard to write bad code in TS |
+| [prettier][] | Code formatter | Never argue about syntax for JS/TS/CSS/JSON |
+| [Jest][] | Test runner | Fast, watches for changes, runs only changed code, etc |
+| [Storybooks][] | Prototyping environment | Make browsable scenes of state + components |
+| [Husky][] | Simplify Git Hooks | Easily ensure code is consistent |
+| [DangerJS][] | Code review automation | Stop asking for CHANGELOG entries etc |
+
+
+</center></article>
+<article class='post'>
+
+These inter-linked, composable tools basically represent the entire idea of the JavaScript community. You add them to your project, and your project gets all these small config files that eventually creates the kind of cohesive tooling that you would expect from a single vendor in a native environment.
+
+The good part is that they are interchangeable, for example, we switched from Flow to TypeScript with roughly 2 week's work, then a week to come close to perfect. The bad side is that the configuration aspects of these projects feels like something you do once, then forget until it needs to change.
+
+I wrote up a glossary of terms from JavaScript when I first started understanding the community, you can [find it here][js-glossary].
+
+*"But JavaScript is such a downgrade compared to Swift"*, I hear you echoing from the back. This is my own perspective but I consider my workflow in TypeScript to be significantly better than working in Swift. My code is statically analyzed as I type, it is auto-styled as I save, my tests are instant, permanently running and show inside my editor at all times, it's portable across platforms, it's quick to execute and embarrassingly fast to compile. I can contribute to all of these tools to improve them. I'd rather work in that environment.
+
+Syntax wise, TypeScript doesn't have enums with associated types, which I like, nor implicit trailing closures. As a trade-off though TypeScript has [union types][union], which work really well for data-modeling. One particular thing I like a lot is that the code you read is so much simpler, as all symbols that you use need to be defined in that file.
```

Comment:
Does this feel better:

> One particular thing I like a lot about JavaScript that the code you read is generally simpler, as nearly all symbols that you use need to be defined in that file. This is because you can only import a single file, instead of an entire target. Meaning if you include a library, you either import the functions you want, or the library as a whole and then extract the functions, and variables that you want.

?